src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.spec.ts

```diff
@@ -1,33 +1,490 @@
 import { HttpClientTestingModule } from '@angular/common/http/testing';
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { FormsModule } from '@angular/forms';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
 import { RouterTestingModule } from '@angular/router/testing';
+import { NgbModal, NgbModule } from '@ng-bootstrap/ng-bootstrap';
+import { of } from 'rxjs';
+import { CourseService } from '../../../services/course.service';
+import { FeedbackQuestionsService } from '../../../services/feedback-questions.service';
+import { FeedbackSessionsService } from '../../../services/feedback-sessions.service';
+import { InstructorService } from '../../../services/instructor.service';
+import { NavigationService } from '../../../services/navigation.service';
+import { StudentService } from '../../../services/student.service';
+import {
+  Course,
+  FeedbackMcqQuestionDetails,
+  FeedbackParticipantType,
+  FeedbackQuestion,
+  FeedbackQuestions,
+  FeedbackQuestionType,
+  FeedbackRankRecipientsQuestionDetails,
+  FeedbackSession,
+  FeedbackSessionPublishStatus,
+  FeedbackSessionSubmissionStatus,
+  FeedbackTextQuestionDetails,
+  FeedbackVisibilityType,
+  Instructor,
+  Instructors,
+  JoinState,
+  NumberOfEntitiesToGiveFeedbackToSetting,
+  ResponseVisibleSetting,
+  SessionVisibleSetting,
+  Student,
+  Students,
+} from '../../../types/api-output';
+import { AjaxLoadingModule } from '../../components/ajax-loading/ajax-loading.module';
+import { CopySessionModalComponent } from '../../components/copy-session-modal/copy-session-modal.component';
+import { QuestionEditFormModel } from '../../components/question-edit-form/question-edit-form-model';
+import { SessionEditFormModel } from '../../components/session-edit-form/session-edit-form-model';
 import { TeammatesRouterModule } from '../../components/teammates-router/teammates-router.module';
 import { InstructorSessionEditPageComponent } from './instructor-session-edit-page.component';
 import { InstructorSessionEditPageModule } from './instructor-session-edit-page.module';
+import Spy = jasmine.Spy;
+
+class MockNgbModalRef {
+  componentInstance: any = {
+    newFeedbackSessionName: '',
+    courseCandidates: [],
+    sessionToCopyCourseId: '',
+  };
+  result: Promise<any> = Promise.resolve();
```

Comment:
Can probably shift this class to within the test case too?
Also we might want to test `copyCurrentSession()` further by spying on say `createFeedbackSession()` for the feedback sessions service. That could be quite involved though, would have to change a couple things and return a promise for `copyCurrentSession()` so we know when it's done. Then this modalRef's `result` would need to give the appropriate desired result also